tests/functional/lti_certification/v13/conftest.py

```diff
@@ -0,0 +1,173 @@
+import json
+from datetime import datetime
+
+import importlib_resources
+import jwt
+import pytest
+from httpretty import httpretty
+
+from tests import factories
+
+keys_path = importlib_resources.files("tests.functional.lti_certification.v13")
+JWT_KID = "JWT_KEY_KID"
+
+
+@pytest.fixture(scope="session")
+def jwt_private_key():
+    with open(keys_path / "jwt_private.key", encoding="utf8") as key:
+        return key.read()
+
+
+@pytest.fixture(scope="session")
+def jwt_public_key():
+    with open(keys_path / "jwt_public.key", encoding="utf8") as key:
+        return json.load(key)
+
+
+@pytest.fixture
+def jwt_headers():
+    return {"kid": JWT_KID}
+
+
+@pytest.fixture
+def make_jwt(jwt_headers, jwt_private_key):
+    def _make_jwt(payload, headers=None):
+        if not headers:
+            headers = jwt_headers
+
+        return jwt.encode(payload, jwt_private_key, algorithm="RS256", headers=headers)
+
+    return _make_jwt
+
+
+@pytest.fixture(autouse=True)
+def lti_registration(db_session):  # pylint:disable=unused-argument
+    return factories.LTIRegistration(
+        issuer="https://ltiadvantagevalidator.imsglobal.org",
+        client_id="imstester_4ba76ab",
+        key_set_url="https://oauth2server.imsglobal.org/jwks",
+        token_url="https://ltiadvantagevalidator.imsglobal.org/ltitool/authcodejwt.html",
+    )
+
+
+@pytest.fixture(autouse=True)
+def application_instance(lti_registration):
+    return factories.ApplicationInstance(
+        tool_consumer_instance_guid="TEST_CONSUMER_INSTANCE_GUID",
+        lti_registration=lti_registration,
+        deployment_id="testdeploy",
+    )
+
+
+@pytest.fixture(autouse=True)
+def jwk_endpoint_intercept(lti_registration, jwt_public_key):
+    jwt_public_key["kid"] = JWT_KID
+    jwk = {"keys": [jwt_public_key]}
```

Comment:
This part is a little confusing to me. If we want `jwt_public_key` to have a fixed `kid` why don't we do it in the `jwt_public_key` fixture? Does it sometimes need to not have this value?

Also, if we want it to match the value in the headers it might be nice to read one from the other. For example, the headers could be:

```python
@pytest.fixture
def jwt_headers(jwt_public_key):
    return {"kid": jwt_public_key["kid"]}
```

Which would:

 * Explain that these two are supposed to line up explicitly
 * Remove the need for the `JWT_KID` variable, as it would be explicitly set inside `jwt_public_key`